koku/reporting/migrations/0102_ocp_all_materialized_views.py

```diff
@@ -0,0 +1,465 @@
+# Generated by Django 2.2.10 on 2020-03-09 22:04
+from django.contrib.postgres.indexes import GinIndex
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [("reporting", "0101_ocpenabledtagkeys"), ("reporting_common", "0021_source_service_product")]
+
+    operations = [
+        # Add index to usage_start on 2 materialized views for ocp_all
+        migrations.RunSQL(
+            sql="""
+DROP INDEX IF EXISTS ocpallcstprjdlysumm_usage_start;
+CREATE INDEX ocpallcstprjdlysumm_usage_start ON reporting_ocpallcostlineitem_project_daily_summary (usage_start);
+
+DROP INDEX IF EXISTS ocpallcstdlysumm_usage_start;
+CREATE INDEX ocpallcstdlysumm_usage_start on reporting_ocpallcostlineitem_daily_summary (usage_start);
+
+DROP INDEX IF EXISTS ocpallcstdlysumm_prod_code_ilike;
+CREATE INDEX ocpallcstdlysumm_prod_code_ilike ON reporting_ocpallcostlineitem_daily_summary USING GIN (upper(product_code) gin_trgm_ops);
+
+DROP INDEX IF EXISTS ocpallcstdlysumm_prod_code_like;
+CREATE INDEX ocpallcstdlysumm_prod_code_like ON reporting_ocpallcostlineitem_daily_summary USING GIN (product_code gin_trgm_ops);
+
+DROP INDEX IF EXISTS ocpallcstdlysumm_prod_family_ilike;
+CREATE INDEX ocpallcstdlysumm_prod_family_ilike ON reporting_ocpallcostlineitem_daily_summary USING GIN (upper(product_family) gin_trgm_ops);
+
+DROP INDEX IF EXISTS ocpallcstdlysumm_prod_family_like;
+CREATE INDEX ocpallcstdlysumm_prod_family_like ON reporting_ocpallcostlineitem_daily_summary USING GIN (product_family gin_trgm_ops);
+            """
+        ),
+        migrations.RemoveIndex(model_name="ocpawscostlineitemdailysummary", name="cost_summary_namespace_idx"),
+        migrations.RemoveIndex(model_name="ocpazurecostlineitemdailysummary", name="ocpazure_namespace_idx"),
+        migrations.AddIndex(
+            model_name="ocpawscostlineitemdailysummary",
+            index=GinIndex(fields=["namespace"], name="cost_summary_namespace_idx"),
+        ),
+        migrations.AddIndex(
+            model_name="ocpazurecostlineitemdailysummary",
+            index=GinIndex(fields=["namespace"], name="ocpazure_namespace_idx"),
+        ),
+        migrations.RunSQL(
+            sql="""
+DROP MATERIALIZED VIEW IF EXISTS reporting_ocpallcostlineitem_daily_summary_database;
+CREATE MATERIALIZED VIEW reporting_ocpallcostlineitem_daily_summary_database AS
```

Comment:
The idea for these materialized views is to precompute the group bys for each API type and reduce down the data returned to just the essentials. The issue referenced a few PRs that had examples for what we did for AWS. Want to walk through those to see what we need here?

Reply:
@adberglund Yes, If I've missed the point, a walk-through will be invaluable.